src/map/skill.c

```diff
@@ -15277,12 +15278,18 @@ static int skill_check_condition_char_sub(struct block_list *bl, va_list ap)
 				return 1;
 			default: //Warning: Assuming Ensemble Dance/Songs for code speed. [Skotlex]
 				{
-					uint16 skill_lv;
 					if(pc_issit(tsd) || !unit->can_move(&tsd->bl))
 						return 0;
+
+					uint16 skill_lv = pc->checkskill(tsd, skill_id);
+#ifdef RENEWAL // In Renewal, partner also gets the requirements consumed, so we must check it
```

Comment:
@skyleo another question, is it correct that in Pre-RE, when we cast an Ensemble skill, the partner doesn't pay the initial SP requirement?
Example: Ring Nibelungen costs 50 SP to be cast + 1 SP/3 second.

If player A casts it, and player B is forced to join it, then:
1. Player A pays 50 SP
2. After 3 seconds: Player A + Player B pays 1 SP
3. ...

Is that right for Pre-RE? or should Player B also pay 50 SP to start the ensemble skill? In Renewal after rebalance, I could observe that this is the behavior, but the existing code didn't consume the requirement from Player B.

Reply:
Yeah this seems to be the behavior in Pre-RE from what I've seen in Ghidra.